janus_core/cli/phonons.py

```diff
@@ -0,0 +1,295 @@
+):
+    """
+    Perform phonon calculations and write out results.
+
+    Parameters
+    ----------
+    ctx : Context
+        Typer (Click) Context. Automatically set.
+    struct : Path
+        Path of structure to simulate.
+    struct_name : Optional[PathLike]
+        Name of structure to simulate. Default is inferred from filepath or chemical
+        formula.
+    supercell : str
+        Supercell lattice vectors. Must be passed in the form '1x2x3'. Default is
+        2x2x2.
+    displacement : float
+        Displacement for force constants calculation, in A. Default is 0.01.
+    thermal : bool
+        Whether to calculate thermal properties. Default is False.
+    temp_start : float
+        Start temperature for CV calculations, in K. Unused if `thermal` is False.
+        Default is 0.0.
+    temp_end : float
+        End temperature for CV calculations, in K. Unused if `thermal` is False.
+        Default is 1000.0.
+    temp_step : float
+        Temperature step for CV calculations, in K. Unused if `thermal` is False.
+        Default is 50.0.
+    band : bool
+        Whether to calculate and save the band structure. Default is False.
+    hdf5 : bool
+        Whether to save force constants in hdf5 format. Default is True.
+    plot_to_file : bool
+        Whether to plot. Default is False.
+    dos : bool
+        Whether to calculate and save the DOS. Default is False.
+    pdos : bool
+        Whether to calculate and save the PDOS. Default is False.
+    minimize : bool
+        Whether to minimize structure before calculations. Default is False.
+    symmetrize : bool
+        Whether to symmetrize force constants. Default is False.
+    fmax : float
+        Set force convergence criteria for optimizer in units eV/Å.
+        Default is 0.1.
+    minimize_kwargs : Optional[dict[str, Any]]
+        Other keyword arguments to pass to geometry optimizer. Default is {}.
+    arch : Optional[str]
+        MLIP architecture to use for geometry optimization.
+        Default is "mace_mp".
+    device : Optional[str]
+        Device to run model on. Default is "cpu".
+    read_kwargs : Optional[dict[str, Any]]
+        Keyword arguments to pass to ase.io.read. Default is {}.
+    calc_kwargs : Optional[dict[str, Any]]
+        Keyword arguments to pass to the selected calculator. Default is {}.
+    file_prefix : Optional[PathLike]
+        Prefix for output filenames. Default is inferred from structure name, or
+        chemical formula.
+    log : Optional[Path]
+        Path to write logs to. Default is "phonons.log".
+    summary : Path
+        Path to save summary of inputs and start/end time. Default is
+        phonons.yml.
+    config : Path
+        Path to yaml configuration file to define the above options. Default is None.
+    """
+    # Check options from configuration file are all valid
+    check_config(ctx)
+
+    [read_kwargs, calc_kwargs, minimize_kwargs] = parse_typer_dicts(
```

Comment:
Square brackets on the left-hand side seems odd, what does this mean?

Reply:
this is from elliott so nee dot check

Reply:
removed them